File: CurrencyConverter/app/src/main/java/com/example/currencyconverter/MainActivity.java
package com.example.currencyconverter;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    public void convertCurrency(View view){
        Log.i("info","Button Pressed");
        EditText editText=(EditText) findViewById(R.id.edNum);
        String amountInPounds=editText.getText().toString();
        double amountInPoundsDouble=Double.parseDouble(amountInPounds);
        double amountInDollerDouble=amountInPoundsDouble*1.3;
        String amountInDollerString=String.format("%.2f",amountInDollerDouble);
        Toast.makeText(this,"#"+amountInPounds+"is $"+ amountInDollerString, Toast.LENGTH_LONG).show();

    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: DiceRoller/app/src/main/java/com/example/diceroller/MainActivity.kt
package com.example.diceroller

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import androidx.databinding.DataBindingUtil
import com.example.diceroller.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private var count:Int=0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
       binding= DataBindingUtil.setContentView(this,R.layout.activity_main)

        binding.button.setOnClickListener{
            rollDice(it)

        }

    }
    private fun rollDice(view: View){
        count++
        val current_value=count%7
        when(current_value){
            1 ->{
                binding.textView.text=current_value.toString();
                binding.imageView.setImageResource(R.drawable.dice_1)

            }
            2 ->{
                binding.textView.text=current_value.toString();
                binding.imageView.setImageResource(R.drawable.dice_2)
            }
            3 ->{
                binding.textView.text=current_value.toString();
                binding.imageView.setImageResource(R.drawable.dice_3)
            }

            4 -> {
                binding.textView.text=current_value.toString();
                binding.imageView.setImageResource(R.drawable.dice_4)
            }
            5 ->{
                binding.textView.text=current_value.toString();
                binding.imageView.setImageResource(R.drawable.dice_5)
            }
            6 -> {
                binding.textView.text=current_value.toString();
                binding.imageView.setImageResource(R.drawable.dice_6)
            }
            else ->
            {
                binding.textView.text=current_value.toString();
                binding.imageView.setImageResource(R.drawable.empty_dice)
            }
        }


    }
}
